packages/sitecore-jss-react/src/components/FEaaSComponent.tsx

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable react/prop-types */
+import React, { useEffect } from 'react';
+import { ComponentParams, ComponentFields, getFieldValue } from '@sitecore-jss/sitecore-jss/layout';
+// eslint-disable-next-line
+import type * as _FEAAS from '@sitecore-feaas/clientside';
+
+export const FEAAS_COMPONENT_RENDERING_NAME = 'FEaaSComponent';
+export const FEAAS_MODULE_SRC = 'https://feaasstatic.blob.core.windows.net/packages/clientside/latest/browser/index.esm.js';
+
+export type FEaaSComponentParams = ComponentParams & {
+  LibraryId?: string;
+  ComponentId?: string;
+  ComponentVersion?: string;
+  ComponentRevision?: string;
+  ComponentHostName?: string;
+  ComponentInstanceId?: string;
+  ComponentDataOverride?: string;
+  ComponentHTMLOverride?: string;
+};
+
+export type FEaaSComponentProps = {
+  params?: FEaaSComponentParams;
+  fields?: ComponentFields;
+};
+
+const getDataFromFields = (fields: ComponentFields): { [key: string]: unknown } => {
+  let data: { [key: string]: unknown } = {};
+  data = Object.entries(fields).reduce((acc, [key]) => {
+    acc[key] = getFieldValue(fields, key);
+    return acc;
+  }, data);
+  return data;
+};
+
+export const FEaaSComponent = ({ params, fields }: FEaaSComponentProps): JSX.Element => {
+  if (
+    !params ||
+    !params.LibraryId ||
+    !params.ComponentId ||
+    !params.ComponentVersion ||
+    !params.ComponentRevision ||
+    !params.ComponentHostName
+  ) {
+    // Missing FEaaS component required props
+    return null;
+  }
+
+  // Load script asynchronously on the page once, only if FEAAS component is used
+  useEffect(() => {
+    window.FEAASLoading ??= new Promise((resolve) => {
```

Comment:
Can this script just be rendered in a component if _FEAASLoading_ is present? Or it's expected that _FEAASLoading_ will be updated once component is rendered?
```html
<script src="" async>
```

Reply:
The idea behind this check is two fold:
1) If there're multiple components on the page, only one will initiate the loading, but others can "listen" to it if they need to
2) If script was bundled instead of lazy loading, this promise instead will be provided by script itself, so in that case it will not load one more time.